great_expectations/execution_engine/sqlalchemy_data_splitter.py

```diff
@@ -0,0 +1,506 @@
+            column_name=column_name,
+            batch_identifiers=batch_identifiers,
+            date_parts=[DatePart.YEAR, DatePart.MONTH, DatePart.DAY],
+        )
+
+    def split_on_date_parts(
+        self,
+        column_name: str,
+        batch_identifiers: dict,
+        date_parts: Union[List[DatePart], List[str]],
+    ) -> Union[BinaryExpression, BooleanClauseList]:
+        """Split on date_part values in column_name.
+
+        Values are NOT truncated, for example this will return data for a
+        given month (if only month is chosen for date_parts) for ALL years.
+        This may be useful for viewing seasonality, but you can also specify
+        multiple date_parts to achieve date_trunc like behavior e.g.
+        year, month and day.
+
+        Args:
+            column_name: column in table to use in determining split.
+            batch_identifiers: should contain a dateutil parseable datetime whose date parts
+                will be used for splitting or key values of {date_part: date_part_value}
+            date_parts: part of the date to be used for splitting e.g.
+                DatePart.DAY or the case-insensitive string representation "day"
+
+        Returns:
+            List of boolean clauses based on whether the date_part value in the
+                batch identifier matches the date_part value in the column_name column.
+        """
+        if len(date_parts) == 0:
+            raise ge_exceptions.InvalidConfigError(
+                "date_parts are required when using split_on_date_parts."
+            )
+
+        column_batch_identifiers: dict = batch_identifiers[column_name]
+        date_parts: List[DatePart] = [
+            DatePart(date_part.lower()) if isinstance(date_part, str) else date_part
+            for date_part in date_parts
+        ]
+
+        if isinstance(column_batch_identifiers, str):
+            column_batch_identifiers: datetime.datetime = parse(
+                column_batch_identifiers
+            )
+
+        if isinstance(column_batch_identifiers, datetime.datetime):
+            query: Union[BinaryExpression, BooleanClauseList] = sa.and_(
+                *[
+                    sa.extract(date_part.value, sa.column(column_name))
+                    == getattr(column_batch_identifiers, date_part.value)
+                    for date_part in date_parts
+                ]
+            )
+        else:
+            query: Union[BinaryExpression, BooleanClauseList] = sa.and_(
+                *[
+                    sa.extract(date_part.value, sa.column(column_name))
+                    == column_batch_identifiers[date_part.value]
+                    for date_part in date_parts
+                ]
+            )
+
+        return query
+
+    def split_on_whole_table(self, batch_identifiers: dict) -> bool:
+        """'Split' by returning the whole table"""
+
+        # return sa.column(column_name) == batch_identifiers[column_name]
+        return 1 == 1
```

Comment:
Should this be updated?

Reply:
Definitely, but I decided to leave these splitters as-is until we had a chance to review in more detail. But happy to revisit in a future effort!

Reply:
You know what - we can always look back in version control if we need to revert this after more consideration. I'll edit.